src/AIAPI/MainGame/GameAPI.Hooks.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections;
+using HarmonyLib;
+using AIProject.SaveData;
+using AIProject;
+using Sirenix.Serialization;
+using AIProject.Scene;
+using Manager;
+
+namespace KKAPI.MainGame
+{
+    public static partial class GameAPI
+    {
+        private class Hooks
+        {
+            public static int lastCurrentDay = 1;//Always starts at 1
+            public static bool isNewGame = false;
+
+            public static void SetupHooks()
+            {
+                Harmony.CreateAndPatchAll(typeof(Hooks));
+            }
+
+            //TODO SaveData.Load only has 1 param in AI and it's the path, does not include filename like KK
+            [HarmonyPostfix]
+            [HarmonyPatch(typeof(SaveData), nameof(SaveData.Load), new[] { typeof(string) })]
+            public static void LoadHook(string fileName)
+            {
+                OnGameBeingLoaded("", fileName);
+            }
+
+            [HarmonyPrefix]
+            [HarmonyPatch(typeof(SaveData), nameof(SaveData.SaveFile), new[] { typeof(string) })]
+            public static void SaveHook(string path)
+            {
+                OnGameBeingSaved(path, "");
+            }
+
+            [HarmonyPrefix]
+            [HarmonyPatch(typeof(TitleLoadScene), "SetWorldData", typeof(WorldData), typeof(bool))]
+            public static void TitleLoadScene_SetWorldData(WorldData _worldData, bool isAuto)
+            {
+                isNewGame = _worldData?.SaveTime.Millisecond == 0;
```

Comment:
I don't think this is right because it could be true of any save (since it's true once per second).

Maybe
```
isNewGame = _wordData?.SaveTime = Game.Instance.Data.AutoData.SaveTime;
```

Reply:
Ah, I assumed it was Total milliseconds.  So yea there would be a 1/1000 chance that it might be incorrectly fired :)  Ill take another stab at it.

Reply:
isNewGame = _worldData?.SaveTime == new DateTime(0);   
Seems to do the trick.

Reply:
That should be fine, yeah.